File: data/supabase_client.py
import os
from supabase import create_client, Client
from dotenv import load_dotenv

# Load environment variables from .env (e.g., ml/.env or project root)
load_dotenv()


def get_client() -> Client:
    """
    Return a Supabase client configured from environment variables.

    Required env vars:
      - SUPABASE_URL
      - SUPABASE_SERVICE_ROLE_KEY
    """
    url = os.environ["SUPABASE_URL"]
    key = os.environ["SUPABASE_ANON_KEY"]
    return create_client(url, key)

File: src/supabase-client.ts
import { createClient } from '@supabase/supabase-js';

const SUPABASE_ANON_KEY = import.meta.env.VITE_SUPABASE_ANON_KEY as string;
const SUPERBASE_URL = import.meta.env.VITE_SUPABASE_URL as string;

export const supabaseClient = createClient(SUPERBASE_URL, SUPABASE_ANON_KEY);
